tracer/src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs

```diff
@@ -290,25 +290,27 @@ private static void SetLegacyResourceNames(BeforeActionStruct typedArg, Span spa
                 routeTemplate = $"{controllerName}/{actionName}";
             }
 
-            string resourceName = $"{httpMethod} {routeTemplate}";
-
-            // override the parent's resource name with the MVC route template
-            span.ResourceName = resourceName;
+            return $"{httpMethod} {routeTemplate}";
         }
 
-        private static Span StartMvcCoreSpan(Tracer tracer, Span parentSpan, BeforeActionStruct typedArg, HttpContext httpContext, HttpRequest request)
+        private static Span StartMvcCoreSpan(
+            Tracer tracer,
+            AspNetCoreHttpRequestHandler.RequestTrackingFeature trackingFeature,
```

Comment:
This is the key change - instead of passing in the currently active span as the "parentSpan" and assuming it's the route span, we use the tracking feature to get the aspnetcore span properly